### PR TITLE
Record expired rdt information

### LINKED ISSUE #114: Save QR code information for Expired RDTs
We will need to save QR barcode information if it is scanned, even if it is not used and returned as an expired RDT.  We would like to collect this information to see how many times the CHW scans an expired RDT.  This should be implemented for v1 release as KEMRI will be using RDTs in the first batch beginning end of August.

### PR DESCRIPTION
**Pull request by vincent-karuri**
closes #114

### CODE REVIEW

File: rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterActivityInteractor.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     * @throws JSONException
+     */
+    public Patient getPatientForRDT(JSONObject jsonFormObject) throws JSONException {
+        Patient rdtPatient = null;
+        if (PATIENT_REGISTRATION.equals(jsonFormObject.optString(ENCOUNTER_TYPE))) {
+            JSONArray formFields = JsonFormUtils.fields(jsonFormObject);
+            JSONObject fieldJsonObject;
+            for (int i = 0; i < formFields.length(); i++) {
+                fieldJsonObject = formFields.getJSONObject(i);
+                if (CONDITIONAL_SAVE.equals(fieldJsonObject.optString(KEY)) &&
```
Comment (githengi) [inline rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterActivityInteractor.java]: What is the purpose of looping through the fields while the `fieldJsonObject` is not used. The field used to populate the patient are not dependent on `fieldJsonObject`.

If testing if the field exists why not use  `FormUtils.getFieldJSONObject(formFields, CONDITIONAL_SAVE)` check if its not null and value is 1

File: rdt-app/src/main/java/io/ona/rdt_app/util/RDTJsonFormUtils.java [hunk 2]
```diff
@@ -82,17 +84,16 @@ protected ProfileImage doInBackground(Void... voids) {
                         profileImage.setSyncStatus(ImageRepository.TYPE_Unsynced);
                         ImageRepository imageRepo = RDTApplication.getInstance().getContext().imageRepository();
                         imageRepo.add(profileImage);
-
-//                        saveImageToGallery(context, image); // todo: only enable this in debug apks
+                        // saveImageToGallery(context, image);
                     }
                 } catch (FileNotFoundException e) {
-                    Log.e(TAG, "Failed to save static image to disk");
+                    Log.e(TAG, e.getStackTrace().toString());
```
Comment (githengi) [inline rdt-app/src/main/java/io/ona/rdt_app/util/RDTJsonFormUtils.java]: Consider to use Timber and log the error directly instead of stackTrace as string

Comment (vincent-karuri) [reply rdt-app/src/main/java/io/ona/rdt_app/util/RDTJsonFormUtils.java]: Done.

File: rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterActivityInteractor.java (hunk 1 above)
Comment (vincent-karuri) [reply rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterActivityInteractor.java]: That works. Done.

**Pull request merged**